### DISCUSSION TITLE
Autocompletion / suggestions partially working

### DISCUSSION THREAD

**Discussion by vonAlenberg** [eclipse-theia/theia#7757] [General] [answered]
I'm new to Theia and have no idea how to fix it... or if it's normal behavior.

**In VSCode:**
* Create a `test.c` file and type `main` => editor suggests a snippet with a complete main function.
* Create a `test.java` file and type `main` => same result.
* Create a `test.tex` file and type `\` => many commands are shown to insert.

**In Theia** (based on the full image, with added Latex Workshop / texlive):
* Create a `test.c` file and type `main` => no suggestion.
* Create a `test.java` file and type `main` => works like in VSCode.
* Create a `test.tex` file and type `\` => no suggestions.

**My questions:**
* Why?
* How to fix?

**Comment by vince-fugnitto** [comment, marked as answer]
@vonAlenberg based on your findings and with a little bit of manual testing, it looks like you are not running the same extensions in **vscode** and **theia**.

> Create a `test.c` file and type `main` => editor suggests a snippet with a complete main function.

The following use-case also does not work in vscode unless the [cpptools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) extension is included. You can verify by running vscode with extensions disabled, and retrying your use-case.

> Create a `test.tex` file and type \ => no suggestions

The following worked for me when using [LaTeX Workshop](https://open-vsx.org/extension/James-Yu/latex-workshop) extension in theia:

<div align='center'>

<img width="1175" alt="Screen Shot 2020-05-06 at 9 16 00 AM" src="https://user-images.githubusercontent.com/40359487/81181373-60bcf600-8f7a-11ea-9885-ef39cde72fc4.png">


</div>

**Comment by vonAlenberg** [comment]
Thanks, for your reply!
That's interesting, because my `latest.pakacges.json` contains 
```
"vscode-builtin-cpp": "https://github.com/theia-ide/vscode-builtin-extensions/releases/download/v1.39.1-prel/cpp-1.39.1-prel.vsix",
``` 
so I thought it would be the same as `cpptools` in VSCode.

![grafik](https://user-images.githubusercontent.com/26161253/81208866-fd5db300-8fcf-11ea-9544-9375e3d28df1.png)


Regarding LaTeX, how did you install the extension? I did via extensions page in theia, so not "pre-bundled" with `package.json`, maybe that's my mistake?

![grafik](https://user-images.githubusercontent.com/26161253/81209073-4ca3e380-8fd0-11ea-86dd-b201cab0e01d.png)

**Reply by vince-fugnitto** [reply]
> That's interesting, because my `latest.pakacges.json` contains
> 
> ```
> "vscode-builtin-cpp": "https://github.com/theia-ide/vscode-builtin-extensions/releases/download/v1.39.1-prel/cpp-1.39.1-prel.vsix",
> ```

No, [`vscode-builtin-cpp`](https://github.com/microsoft/vscode/tree/master/extensions/cpp) is the builtin C/C++ support from vscode which contributes syntax-highlighting mostly. Additional extensions are required to have richer C/C++ support.


> Regarding LaTeX, how did you install the extension? I did via extensions page in theia, so not "pre-bundled" with `package.json`, maybe that's my mistake?
> 

I initially added it to my `package.json` but it's also not working properly for me.
If it is still the case for you please open an issue with steps to reproduce.
